crates/core/c8y_api/src/smartrest/inventory.rs

```diff
@@ -44,15 +51,41 @@ pub fn service_creation_message(
     service_type: &str,
     service_status: &str,
     ancestors: &[String],
-) -> Message {
-    Message::new(
+) -> Result<Message, InvalidValueError> {
+    // TODO: most of this noise can be eliminated by implementing `Serialize`/`Deserialize` for smartrest format
```

Comment:
As you said here, why not implementing the representation struct for SmartREST 101 and 102? We already have them for other message IDs, for operations. 
https://github.com/thin-edge/thin-edge.io/blob/main/crates/core/c8y_api/src/smartrest/smartrest_serializer.rs